src/s390x/dasd.rs

```diff
@@ -0,0 +1,284 @@
+        .arg("cdl")
+        .arg("--mode")
+        .arg("full")
+        .arg("-y")
+        .arg("-p")
+        .arg("-v")
+        .arg(dasd)
+        .stdout(std::process::Stdio::inherit())
+        .status()
+        .chain_err(|| "failed to execute dasdfmt")?;
+    check(status).chain_err(|| "low-level format")
+}
+
+/// Format disk and create partitions
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+/// * `partitions` - configuration strings
+fn make_partitions(dasd: &str, partitions: &[String]) -> Result<()> {
+    let config_filename = "/tmp/fdasd.conf";
+
+    write(config_filename, partitions.join("\n"))
+        .chain_err(|| format!("writing {}", config_filename))?;
+
+    if try_format(dasd, &config_filename).is_err() {
+        default_format(dasd)?;
+        try_format(dasd, &config_filename)?;
+    }
+    Ok(())
+}
+
+/// Sometimes it may happen, that config-based format fails, so we have to perform
+/// an auto-format on the whole disk
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+fn default_format(dasd: &str) -> Result<()> {
+    eprintln!("Auto-formatting {}", dasd);
+    let status = Command::new("fdasd")
+        .arg("-a")
+        .arg("-r")
+        .arg(dasd)
+        .stdout(std::process::Stdio::inherit())
+        .status()
+        .chain_err(|| "failed to execute fdasd")?;
+    check(status).chain_err(|| "formatting using entire disk")
+}
+
+/// Format disk using a config file
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+/// * `file` - configuration file
+fn try_format<P: AsRef<Path>>(dasd: &str, file: P) -> Result<()> {
+    eprintln!("Formatting {} ", dasd);
+    let status = Command::new("fdasd")
+        .arg("--config")
+        .arg(file.as_ref().to_str().unwrap())
+        .arg(dasd)
+        .stdout(std::process::Stdio::inherit())
+        .status()
+        .chain_err(|| "failed to execute fdasd")?;
+    check(status).chain_err(|| format!("formatting based on {}", file.as_ref().display()))
+}
+
+/// Copy each partition from CoreOS raw image to corresponding /dev/<disk-name>
+///
+/// # Arguments
+/// * `dds` - list of dd options.
+fn copy_partitions(dds: &[Opts]) -> Result<()> {
+    for opts in dds {
+        match dd(&opts) {
+            Ok(s) => {
+                eprintln!("{}", s);
+            }
+            Err(e) => {
+                return Err(e.to_string().into());
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Return disk's "blocks per track" and "bytes per block" config values as tuple
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+fn disk_config(dasd: &str) -> Result<(u64, u64)> {
+    let find = |data: &str, pat: &str| -> Result<u64> {
+        let s = data
+            .find(pat)
+            .chain_err(|| format!("cannot find: \'{}\'", pat))?
+            + pat.len();
+        let e = data[s..].find('\n').chain_err(|| "cannot find newline")? + s;
```

Comment:
Might make more sense to use `regex` here?

Reply:
what's about making a separate ISSUE/patch and replace all manual string processing with regex if/when required?

Reply:
I think it'd be better to do things cleanly from the start.  Is there some reason regexes would be difficult to apply here?

Reply:
np, now regex are in use